### PR TITLE
Feature/wallet modals

### PR DESCRIPTION
**Pull request by jalarg**
# Summary
- Add modal for sending Lummens.
- Add modal for receiving Lummens .
- Refactor custom modal to be usable for the wallet and home modals.
- Add TypeScript interfaces for each modal.

# Details
- This PR introduces two modals, one for sending and other for receiving Lummens, providing users with a streamlined process for cryptocurrency transactions. Additionally, it refactors the existing custom modal to be adaptable for use in both wallet and home modals, improving code reusability.

# Evidence 

## Wallet modals for wallet: 

Modal1

Modal2

Tests

### CODE REVIEW

File: src/components/modals/CustomModal.tsx [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
+import React from "react";
+import StepOneRegistration from "./home/StepOneRegistration";
+import StepTwoRegistration from "./home/StepTwoRegistration";
+import StepOneLogin from "./home/StepOneLogin";
+import StepTwoLogin from "./home/StepTwoLogin";
+import SendTransaction from "./wallet/SendTransaction";
+import ReceiveTransaction from "./wallet/ReceiveTransaction";
 
-const modalComponents: Record<string, React.FC<IModal>> = {
+const modalComponents: Record<string, React.FC<any>> = {
```
Comment (Genebson) [inline src/components/modals/CustomModal.tsx]: Why any?

File: src/components/modals/CustomModal.tsx [hunk 2]
```diff
@@ -1,29 +1,36 @@
...
   label,
-  setPublicKey,
-  setSecretKey,
-  publicKey,
-  secretKey,
-}) => {
+  modalProps,
+}: ModalProps<T>) => {
   const Component = modalComponents[label];
+  console.log(label, "label");
```
Comment (Genebson) [inline src/components/modals/CustomModal.tsx]: Delete this, please.

File: src/components/modals/wallet/ReceiveTransaction.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import copy from "../../../assets/copy-icon.svg";
+import { IWalletModalReceiveTransaction } from "../../../types/types";
+
+const ReceiveTransaction: React.FC<IWalletModalReceiveTransaction> = ({
+  label,
+  onClose,
+  content,
+  publicKey,
+}) => {
+  const [qrValue, setQrValue] = React.useState<string>("");
```
Comment (Genebson) [inline src/components/modals/wallet/ReceiveTransaction.tsx]: This should work
```suggestion
  const [qrValue, setQrValue] = useState<string>("");
```

File: src/components/modals/wallet/ReceiveTransaction.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      <div className="flex flex-col items-center justify-center text-center text-xs sm:text-xl">
+        <div className="flex flex-col items-center justify-center bg-white rounded-[10px] border border-[2] border-gray-600 p-3">
+          <QRCode className="modal-receive-qr" value={qrValue} size={128} />
+        </div>
+      </div>
+      <div className="flex flex-col items-center justify-center">
+        <h3 className={`modal-publicKey-${label} text-xsm sm:text-xsm`}>
+          {`${publicKey?.trim().slice(0, 5)}...${publicKey
+            ?.trim()
+            .slice(publicKey.length - 6, publicKey.length - 1)}`}
```
Comment (Genebson) [inline src/components/modals/wallet/ReceiveTransaction.tsx]: Save this in a function, please.

File: src/components/modals/wallet/ReceiveTransaction.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        </h3>
+        <CopyToClipboard    
+          text={qrValue}
+          onCopy={() => alert("Text copied to the clipboard")}
+        >
+          <Image  src={copy} alt="copy" className="modal-receive-copy-icon m-3 w-4 h-4 cursor-pointer" />
```
Comment (Genebson) [inline src/components/modals/wallet/ReceiveTransaction.tsx]:
```suggestion
          <Image src={copy} alt="copy" className="modal-receive-copy-icon m-3 w-4 h-4 cursor-pointer" />
```

File: src/components/modals/wallet/SendTransaction.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  const handleSetTransactionId = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+  };
+
+  const handleSetAmount = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setAmount(e.target.value);
+  };
+
+  const sendTransactionHandler = () => {
+    console.log(`Transaction sent from ${publicKey} to ${value}`);
```
Comment (Genebson) [inline src/components/modals/wallet/SendTransaction.tsx]: We should use a popup to show this transaction information.

File: src/content/modals.ts [hunk 7]
```diff
@@ -48,3 +48,14 @@ export const loginContentStep2 = {
   ],
   button: "Connect",
 };
+
+export const sendTransaction = {
+    title: "Send Lummens",
```
Comment (Genebson) [inline src/content/modals.ts]:
```suggestion
    title: "Send Lumens",
```

File: src/content/modals.ts [hunk 8]
```diff
@@ -48,3 +48,14 @@ export const loginContentStep2 = {
...
+    button: "Send",
+};
+
+export const receiveTransaction = {
+  title: "Receive Lummens",
```
Comment (Genebson) [inline src/content/modals.ts]:
```suggestion
  title: "Receive Lumens",
```

File: src/pages/wallet/index.tsx [hunk 9]
```diff
@@ -5,14 +5,57 @@ import Footer from "../../components/Footer";
...
+import CustomModal from "../../components/modals/CustomModal";
+import {
+  IWalletModalSendTransaction,
+  IWalletModalReceiveTransaction,
+} from "../../types/types";
 
 function Wallet() {
-  const [publicKey, setPublicKey] = useState<string>("");
+  const [publicKey, setPublicKey] = useState<string>(
+    "G54325GDGFFDGFDGSASDASDASDFGDSFDA87ADAS9D7SADA9D7AS"
```
Comment (Genebson) [inline src/pages/wallet/index.tsx]: Why is this publicKey hardcoded?

File: src/components/modals/home/StepOneLogin.tsx [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+  label,
+  isOpen,
+  onClose,
+  openModal,
+}) => {
+  const [isChecked, setIsChecked] = useState<boolean>(false);
+
+  const handleCancelChecked = () => {
+    setIsChecked(false);
+  };
```
Comment (Genebson) [inline src/components/modals/home/StepOneLogin.tsx]: Can we save this function in an utils.ts file? That way we can import it where we need it.

File: src/components/modals/home/StepTwoRegistration.tsx [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+  content,
+  label,
+  isOpen,
+  onClose,
+  secretKey,
+  publicKey,
+}) => {
+  const [isChecked, setIsChecked] = useState<boolean>(false);
+
+  const handleCancelChecked = () => {
```
Comment (Genebson) [inline src/components/modals/home/StepTwoRegistration.tsx]: Same here

**Review by Genebson — APPROVED** [review]

File: src/components/modals/home/StepOneLogin.tsx (hunk 10 above)
Comment (jalarg) [reply src/components/modals/home/StepOneLogin.tsx]: solved in commit 424537b8225447b094604e1bc6f94afd41adf270.

File: src/components/modals/home/StepTwoRegistration.tsx (hunk 11 above)
Comment (jalarg) [reply src/components/modals/home/StepTwoRegistration.tsx]: solved in commit 424537b8225447b094604e1bc6f94afd41adf270.

**Pull request merged**